pkg/engine/task/pod_exec.go

```diff
@@ -0,0 +1,232 @@
+package task
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strconv"
+
+	"github.com/spf13/afero"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+// PodExec defines a command that will be executed in a running Pod.
+// RestCfg - The REST configuration for the cluster.
+// PodName - The pod name on which to execute the command.
+// PodNamespace - Namespace of the pod
+// ContainerName - optional container to execute the command in. If empty, first container is taken
+// Args - The command (and args) to execute.
+// In - Command input stream.
+// Out - Command output stream
+// Err - Command error stream
+type PodExec struct {
+	RestCfg       *rest.Config
+	PodName       string
+	PodNamespace  string
+	ContainerName string
+	Args          []string
+	In            io.Reader
+	Out           io.Writer
+	Err           io.Writer
+	TTY           bool
+}
+
+// Run executes a command in a pod. This is a distilled version of what `kubectl exec` (and
+// also `kubectl  cp`) doing under the hood: a POST request is made to the `exec` subresource
+// of the v1/pods endpoint containing the pod information and the command. Here is a good article
+// describing it in detail: https://erkanerol.github.io/post/how-kubectl-exec-works/
+func (pe *PodExec) Run() error {
+	codec := serializer.NewCodecFactory(scheme.Scheme)
+	restClient, err := apiutil.RESTClientForGVK(
```

Comment:
hmm why do we need a new client here? why can't we use the client from manager? that one cannot do these kinds of requests? 🤔

Reply:
yeah, the "usual" client does not expose a REST interface. Apparently RESTClientForGVK method  is a [way to go](https://github.com/kubernetes-sigs/controller-runtime/issues/680)